#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_cups_genppdupdate_filename_metacharacters.dpatch by  <rleigh@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix breaking on shell metacharacters in PPD filenames

@DPATCH@

--- gutenprint-4.3.99+cvs20050801.orig/src/cups/cups-genppdupdate.in
+++ gutenprint-4.3.99+cvs20050801/src/cups/cups-genppdupdate.in
@@ -93,7 +93,7 @@
 my @ppdglob = glob("$ppd_dir/*.{ppd,PPD}");
 my $ppdlist = join ' ', @ppdglob;
 if (@ppdglob) {
-    open PPDFILES, "egrep -i -l \"Gutenprint|Gimp-Print\" $ppdlist|" or die "can't grep $ppd_dir/*: $!";
+    open PPDFILES, '-|', 'egrep', '-i', '-l', 'Gutenprint|Gimp-Print', @ppdglob or die "can't grep $ppd_dir/*: $!";
     while (<PPDFILES>) {
 	chomp;
 	push @ppd_files,  $_;
